bson/doc.go

```diff
@@ -117,6 +117,9 @@
 //     can be used with fields that are pointers to structs. If an inlined pointer field is nil, it will not be marshalled.
 //     For fields that are not maps or structs, this tag is ignored.
 //
+// This tag behavior is configurable, and different struct tag behavior can be configured by initializing a new
```

Comment:
We should explicitly mention that we don't honor JSON struct tags by default, but that behavior can be enabled by using this new parser. There should also be a small code sample showing how to create a new StructCodec with this parser.

Reply:
done

Reply:
1. Avoid using "we" in documentation.
2. I'd move this to be before the list of tags where we say `The encoding of each struct field can be customized by the "bson" struct tag.` We can modify that to say that the driver can be configured to respect `json` tags as well via this new parser.